src/runtime/HalideRuntime.h

```diff
@@ -685,25 +685,23 @@ typedef enum halide_target_feature_t {
 
     halide_target_feature_user_context = 24,  ///< Generated code takes a user_context pointer as first argument
 
-    halide_target_feature_register_metadata = 25,  ///< Generated code registers metadata for use with halide_enumerate_registered_filters
+    halide_target_feature_matlab = 25,  ///< Generate a mexFunction compatible with Matlab mex libraries. See tools/mex_halide.m.
```

Comment:
We need to make a call on how we handle this. Theoretically these are now a binary interface and changing the values might break clients. I don't really have a problem insisting everything get recompiled, but I also don't have an issue putting in a comment line here saying what the value used to be and not renumbering.